### PR TITLE
Add controller methods for handling Media Editor plugins

### PR DESCRIPTION
**Pull request by yireo**
See also the sample Media Editor plugin: https://github.com/yireo/plg_media-editor_example

### CODE REVIEW

File: administrator/components/com_media/controllers/file.php [hunk 1]
```diff
@@ -308,4 +310,107 @@ public function delete()
...
+		$postUrl = 'index.php?option=com_media&task=file.editorPost&plugin=' . $pluginName;
+		$html    = $plugin->onMediaEditorDisplay($filePath, $postUrl);
+
+		// @todo: Create actually a view to generate a HTML container for this $html
+		echo $html;
+
+		$app->close();
+	}
+
+	/**
```
Comment (roland-d) [inline administrator/components/com_media/controllers/file.php]: Please fix docbloc

File: administrator/components/com_media/controllers/file.php [hunk 2]
```diff
@@ -308,4 +310,107 @@ public function delete()
...
+
+		$app = JFactory::getApplication();
+
+		$file       = $app->input->getString('file');
+		$pluginName = $app->input->getCmd('plugin');
+		$plugin     = $this->loadPlugin($pluginName);
+
+		if ($plugin == false)
+		{
+			throw new RuntimException('Unsupported plugin');
```
Comment (roland-d) [inline administrator/components/com_media/controllers/file.php]: This needs to be a language string

File: administrator/components/com_media/controllers/file.php [hunk 3]
```diff
@@ -308,4 +310,107 @@ public function delete()
...
+	protected function loadPlugin($pluginName)
+	{
+		$pluginData = JPluginHelper::getPlugin('media-editor', $pluginName);
+
+		if (empty($pluginData))
+		{
+			return false;
+		}
+
+		$fileName = JPATH_ROOT . '/plugins/media-editor/' . $pluginData->name . '/' . $pluginData->name . '.php';
```
Comment (roland-d) [inline administrator/components/com_media/controllers/file.php]: Should we check if the file actually exists?

File: administrator/components/com_media/controllers/file.php [hunk 4]
```diff
@@ -308,4 +310,107 @@ public function delete()
...
+		$plugin     = new $className($dispatcher, (array) $pluginData);
+
+		if (!$plugin instanceof JPlugin)
+		{
+			return false;
+		}
+
+		return $plugin;
+	}
+}
```
Comment (roland-d) [inline administrator/components/com_media/controllers/file.php]: Files must end with an empty line

File: administrator/components/com_media/models/file/adapter/local.php [hunk 5]
```diff
@@ -60,18 +60,4 @@ public function getMimeType()
...
-	 *
-	 * @return $this
-	 */
-	public function setFilePath($filePath)
-	{
-		$this->filePath = $filePath;
-
-		return $this;
-	}
 }
```
Comment (roland-d) [inline administrator/components/com_media/models/file/adapter/local.php]: Empty line is needed at the end of the file

File: administrator/components/com_media/controllers/file.php [hunk 6]
```diff
@@ -308,4 +310,107 @@ public function delete()
...
+		}
+
+		$filePath = JPATH_ROOT . '/' . $file;
+		$plugin->onMediaEditorProcess($filePath);
+
+		$app->close();
+	}
+
+	/**
+	 * Helper method to load a specific Media Editor plugin from its name
```
Comment (roland-d) [inline administrator/components/com_media/controllers/file.php]: Please fix docbloc

File: administrator/components/com_media/controllers/file.php [hunk 7]
```diff
@@ -308,4 +310,107 @@ public function delete()
 
 		return $ret;
 	}
-}
+
+	/**
```
Comment (roland-d) [inline administrator/components/com_media/controllers/file.php]: Please fix docbloc

**Comment by yireo** [comment]
Could somebody submit this PR, so we can sync progress?

**Comment by roland-d** [comment]
@yireo It can't be merged, there are conflicts. Can you fix those?

**Comment by yireo** [comment]
So that's the problem of not being able to commit changes - people start working on the same files. I've removed the conflicts now. Could you try to merge it again?

**Comment by roland-d** [comment]
@yireo merged :)

**Pull request merged**